### PR TITLE
DOC: fix typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell` and `grep`

### CODE REVIEW

**Comment by ilayn** [comment]
I think the `nonlin.py` fixes are not necessary. We don't need to touch the code. Also I'm not a big fan of touching the 3rd party code such as SuperLU or others just for the sake of typos. It might be better fixed at their own code base instead.

File: scipy/optimize/nonlin.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-r"""
+"""
```
Comment (luzpaz) [inline scipy/optimize/nonlin.py]: this rogue `r` does this not effect the code in any way?

Comment (ilayn) [reply scipy/optimize/nonlin.py]: Have a look at this one https://stackoverflow.com/questions/2081640/what-exactly-do-u-and-r-string-flags-do-and-what-are-raw-string-literals

Comment (person142) [reply scipy/optimize/nonlin.py]: It indicates that the docstring is a Python raw string.

**Comment by pv** [comment]
... and it should not be removed here, because there are slashes in the 
string.

**Pull request merged**